File: greedy/big_number/big_number2.java
package greedy.big_number;

import java.util.*;

public class big_number2 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int m = sc.nextInt();
        int k = sc.nextInt();

        int[] data = new int[n];
        for (int i = 0; i < n; i++) {
            data[i] = sc.nextInt();
        }

        Arrays.sort(data);
        int first = data[n - 1];
        int second = data[n - 2];

        int result = 0;

        while (m > 0) {
            for (int i = 0; i < k; i++) {
                if (m == 0)
                    break;
                result = result + first;
                m--;
            }
            if (m == 0)
                break;
            result = result + second;
            m--;
        }

        System.out.println(result);
    }

}

File: greedy/big_number/big_number2.py
# n, m, k 입력 받기
n, m, k = map(int, input().split())

data = list(map(int, input().split()))  # data 입력 받기
data.sort()  # data 정렬

first = data[n - 1]  # 가장 큰 수
second = data[n - 2]  # 두번째로 큰 수
result = 0  # 결과

while m > 0:  # m이 0보다 클 때동안
    for i in range(k):  # k번 반복
        if m == 0:  # m이 0이면
            break  # 반복문 빠져나오기
        result = result + first  # 가장 큰 수 더하기
        m = m - 1  # m 감소
    if m == 0:
        break
    result = result + second  # 두번째로 큰 수 더하기
    m = m - 1

print(result)
